beta-src/src/classes/map/TerritoryMapDataGenerator.ts

```diff
@@ -0,0 +1,103 @@
+import { TerritoryMapData } from "../../interfaces";
+import webDiplomacyTheme from "../../webDiplomacyTheme";
+import Country from "../../enums/Country";
+
+interface TerritoryMapDataGeneratorInterface extends TerritoryMapData {
+  country?: Country;
+}
+
+export default class TerritoryMapDataGenerator
+  implements TerritoryMapDataGeneratorInterface
+{
+  public name: TerritoryMapData["name"];
+
+  public abbr: TerritoryMapData["abbr"];
+
+  public type: TerritoryMapData["type"];
+
+  public centerPos: TerritoryMapData["centerPos"];
+
+  public labels: TerritoryMapData["labels"];
+
+  public unitSlot: TerritoryMapData["unitSlot"];
+
+  public height: TerritoryMapData["height"];
+
+  public width: TerritoryMapData["width"];
+
+  public x: TerritoryMapData["x"];
+
+  public y: TerritoryMapData["y"];
+
+  public path: TerritoryMapData["path"];
+
+  public fill: TerritoryMapData["fill"];
+
+  public texture: TerritoryMapData["texture"];
+
+  public viewBox: TerritoryMapData["viewBox"];
+
+  constructor({
+    name,
```

Comment:
if you put `public name: TerritoryMapData["name"],` in the constructor, you don't have to make the this.name assignments and it greatly reduces the amount of boilerplate code in this file